### ISSUE TITLE
Own group message does not appear in chatlog

### ISSUE THREAD

**Issue by anthonybilinski** [qTox/qTox#5481]
##### Brief Description

OS: Windows 10 and Ubuntu 18.04
qTox version: v1.16.3 and v1.16.3+
Commit hash:  v1.16.3 and aa83edf7599acafcfb546eb02d18834aeb6efa20
toxcore: v0.2.3 and v0.2.8
Qt:  5.9.6 and 5.9.5
Hardware:  Desktops

Reproducible: Rarely

##### Steps to reproduce

1. Join a group, either by invite or by persistent group load.
2. Send message.

##### Observed Behavior
1. Own message does not appear in chatlog.
2. No error is shown or logged.
3. Peers in the group still receive the message.

##### Expected Behavior
1. Own message is displayed in chatlog.

**Comment by anthonybilinski** [comment]
Toxcore appears to deliver our own group messages to ourselfs, leading us to have the behaviour in groupchatform to not normally manually add our own group message, since we rely on receiving the message to add it. It's possible we aren't receiving our own message from Toxcore.

**Comment by anthonybilinski** [comment]
Confirmed that we're calling down to `tox_conference_send_message`, getting back no error, and then not receiving the expected `tox_callback_conference_message` callback. I noticed that disconnecting from the group (by stopping too long in gdb) and then reconnecting fixes the issue for the group. Looks like an upstream issue. I've also confirmed that all peers in the group still get the message.
